File: WHC_KeyboradManager(OC)/WHC_KeyboradManager(OC)/StyleVC2.m
//
//  StyleVC2.m
//  WHC_KeyboardManager(OC)
//
//  Created by WHC on 16/11/26.
//  Copyright © 2016年 WHC. All rights reserved.
//

#import "StyleVC2.h"
#import "WHC_StackView.h"
#import "WHC_KeyboardManager.h"

@interface StyleVC2 ()
@property (weak, nonatomic) IBOutlet UIScrollView *scrollView;
@property (nonatomic, strong)WHC_StackView * stackView;
@end

@implementation StyleVC2

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    self.navigationItem.title = @"ScrollView无键盘头";
    self.view.backgroundColor = [UIColor whiteColor];
    
    self.edgesForExtendedLayout = UIRectEdgeNone;
    
    [_scrollView whc_AutoSize:0 top:0 right:0 bottom:0];
    /// 键盘处理配置
    /*******只需要在要处理键盘的界面创建WHC_KeyboardManager对象即可无需任何其他设置*******/
    WHC_KBMConfiguration * configuration = [[WHC_KeyboardManager share] addMonitorViewController:self];
    /// 不要键盘头
    configuration.enableHeader = false;
    
    /********************* 构建UI ***********************/
    _stackView = [WHC_StackView new];
    /// 设置垂直布局
    _stackView.whc_Orientation = Vertical;
    /// 设置分割线高度
    _stackView.whc_SegmentLineSize = 0.5;
    /// 设置子视图高度
    _stackView.whc_SubViewHeight = 40;
    /// 设置子视图内边距
    _stackView.whc_Edge = UIEdgeInsetsMake(0, 16, 0, 16);

    [_scrollView addSubview:_stackView];
    _stackView.whc_LeftSpace(0)
    .whc_TopSpace(0)
    .whc_RightSpaceToView(0, self.view)
    .whc_HeightAuto()
    .whc_BottomSpaceKeepHeightConstraint(0,YES);/// scrollview contentSize自动
    
    for (int i = 0; i < 30; i++) {
        if (i < 15) {
            UITextView * text = [UITextView new];
            text.text = [NSString stringWithFormat:@"UITextView %d",i];
            text.font = [UIFont systemFontOfSize:14];
            [_stackView addSubview: text];
        }else {
            UITextField * text = [UITextField new];
            text.placeholder = [NSString stringWithFormat:@"UITextField %d",i];
            [_stackView addSubview: text];
        }
    }
    [_stackView whc_StartLayout];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: WHC_KeyboradManager/StyleVC2.swift
//
//  StyleVC2.swift
//  WHC_KeyboradManager
//
//  Created by WHC on 16/11/17.
//  Copyright © 2016年 WHC. All rights reserved.
//

import UIKit

class StyleVC2: UIViewController {

    deinit {
        print("StyleVC2")
    }
    
    @IBOutlet weak var scrollView: UIScrollView!
    private lazy var stackView: WHC_StackView = WHC_StackView()
    private var testModel: TestModel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        testModel = TestModel(self)
        // Do any additional setup after loading the view.
        self.navigationItem.title = "ScrollView无键盘头"
        self.view.backgroundColor = UIColor.white
        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "done", style: .plain, target: self, action: #selector(clickRight(sender:)))
        
        self.navigationItem.leftBarButtonItem = UIBarButtonItem(title: "done", style: .plain, target: self, action: #selector(clickLeft(sender:)))
        /// 键盘处理配置
        /*******只需要在要处理键盘的界面创建WHC_KeyboradManager对象即可无需任何其他设置*******/
        let configuration = WHC_KeyboardManager.share.addMonitorViewController(self)
        /// 不要键盘头
        configuration.enableHeader = false
        
        /********************* 构建UI ***********************/
        /// 设置垂直布局
        stackView.whc_Orientation = .vertical
        /// 设置分割线高度
        stackView.whc_SegmentLineSize = 0.5
        /// 设置子视图高度
        stackView.whc_SubViewHeight = 40
        /// 设置子视图内边距
        stackView.whc_Edge = UIEdgeInsetsMake(0, 16, 0, 16)
        
        scrollView.addSubview(stackView)
        stackView.whc_Left(0)
        .whc_Top(0)
        .whc_Right(0, toView: self.view)
        .whc_HeightAuto()
        .whc_Bottom(0)/// scrollview contentSize自动
        
        for i in 0 ..< 30 {
            if i < 15 {
                let text = UITextView()
                text.text = "UITextView = \(i)"
                text.font = UIFont.systemFont(ofSize: 14)
                stackView.addSubview(text)
            }else {
                let text = UITextField()
                text.placeholder = "UITextField \(i)"
                stackView.addSubview(text)
            }
        }
        stackView.whc_StartLayout()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @objc func clickLeft(sender: UIBarButtonItem) {
        self.dismiss(animated: true, completion: nil)
    }
    
    @objc func clickRight(sender: UIBarButtonItem) {
        self.view.endEditing(true)
        let vc = TestVC(nibName: "TestVC", bundle: nil)
        vc.view.backgroundColor = UIColor(white: 0, alpha: 0.7)
        let currentVC = self.whc_CurrentViewController()
        
//        currentVC?.definesPresentationContext = true
        let nv = UINavigationController(rootViewController: vc)
        nv.modalPresentationStyle = .overCurrentContext
        currentVC?.present(nv, animated: true, completion: nil)
//        currentVC?.present(UINavigationController(rootViewController: vc), animated: true, completion: nil)
//        self.navigationController?.pushViewController(vc, animated: true)
    }

}
